src/index.ts

```diff
@@ -2,24 +2,59 @@
 
 import { Command } from 'commander';
 
+import { loadTolgeeRc, loadAuthToken } from './config';
+
 import Client from './client';
 import { HttpError } from './client/errors';
-import { setDebug, error } from './utils/logger';
+import { setDebug, warn, error } from './utils/logger';
 import { VERSION } from './utils/constants';
 
+import { API_KEY_OPT } from './options';
+import { Login, Logout } from './commands/login';
 import PushCommand from './commands/push';
 import PullCommand from './commands/pull';
 import ExtractCommand from './commands/extract';
 
-function preHandler(prog: Command, cmd: Command) {
-  const opts = cmd.opts();
-  // Validate --project-id and --api-key if necessary
-  if ('apiKey' in opts) {
-    if (opts.project === -1 && !opts.apiKey.startsWith('tgpak_')) {
-      error('Project ID is required when not using a Project API Key.');
+function hasApiKeyArg(cmd: Command): boolean {
+  for (const opt of cmd.options) {
+    if (opt === API_KEY_OPT) {
+      return true;
+    }
+  }
+
+  return cmd.parent ? hasApiKeyArg(cmd.parent) : false;
+}
+
+async function preHandler(prog: Command, cmd: Command) {
+  const opts = cmd.optsWithGlobals();
+
+  if (hasApiKeyArg(cmd)) {
+    if (!opts.apiKey) {
+      // Attempt to load --api-key from config store if not specified
+      // This is not done as part of the init routine or via the mandatory flag, as this is dependent on the API URL.
+      const key = await loadAuthToken(opts.apiUrl);
```

Comment:
We can actually store also multiple PAKs for single apiUrl. It's possible to parse the PAK and read the projectID from it. It's actually "<project_id>_<legacyApikey>" encoded to base32. Check out io.tolgee.service.security.ApiKeyService#parseApiKey of Tolgee Platform.